### DISCUSSION TITLE
Shell Integration is broken on macOS

### DISCUSSION THREAD

**Discussion by ciscohack** [kovidgoyal/kitty#6759] [Q&A]
Hello experts,

I am running the latest Kitty terminal version on macOS 14 and experiencing an issue with shell_integration as per the document shell_integration is automatic but I tried both auto and manual integration nothing is working on macOS Due to this some of the cool features like clone tab and view last command ..jump prompt is not working.  I have tried some community suggestions as well but that also has not helped me to fix the problem. Can someone who os running kitty on macOS and shell_integration working then please help me with configuration? btw I am running ZSH shell for reference.

**Comment by kovidgoyal** [comment]
Run kitty like this:

kitty --config=NONE

type ls -l at the prompt and press enter

Press ctrl+shift+g

you will be viewing the ls output inside the less program.

**Reply by ciscohack** [reply]
Thanks @kovidgoyal for your prompt response. I already tried this step but it's not working

**Reply by ciscohack** [reply]
if i press ctrl+shift+h that is working all time but ctrl+shift+g or ctrl+shift+x,z not working

below is ctrl+shift+h output looks weird but working

<img width="848" alt="image" src="https://github.com/kovidgoyal/kitty/assets/11479557/062b5c8b-16ff-4533-a968-6b6753135bc4">

**Reply by kovidgoyal** [reply]
EMpty your zsh rc files and see if it makesa difference. And from that screenshot the version of less in your PATH is not correct. Presumably you have broken versions of both less and zsh somewhere on your system. Maybe homebrew/macports or similar. Uninstall them and you should be fine.

**Reply by ciscohack** [reply]
@kovidgoyal Here is my running version. btw how did you find that my LESS version has not been shared yet and why do you think it's broken? I have installed it through brew only and the ZSh version I am using comes with macOS by default 

 which less                                                                                                                 
/opt/homebrew/bin/less

less 643 (PCRE2 regular expressions)
Copyright (C) 1984-2023  Mark Nudelman

less comes with NO WARRANTY, to the extent permitted by law.
For information about the terms of redistribution,
see the file named README in the less distribution.
Home page: https://greenwoodsoftware.com/less

**Comment by kovidgoyal** [comment]
Because of the escaping in the screenshot of ctrl+shift+h normal less
does not do that.

As I said, empty your zsh config if that doesnt work, then I am out of
ideas, good luck.

**Reply by ciscohack** [reply]
Lol, hehe loved the last line good luck.. Genius Kovid gave up... any way thanks a lot for your time and help.. I don't know but some common stuff gets so complex in modern terminals

**Comment by ciscohack** [comment]
Here is the output of ctrl + shift + h with no zshrc config but sadly ctrl + shift + g still not working .. in fact I am pretty sure that shell integration is broken on macOS

<img width="1091" alt="image" src="https://github.com/kovidgoyal/kitty/assets/11479557/a544fbcc-522b-47f0-81af-87aeebf46d53">

**Reply by ciscohack** [reply]
you are seeing the escape sequence earlier might be due to the icons in front of files
